yt/data_objects/static_output.py

```diff
@@ -1115,41 +1116,48 @@ def set_units(self):
         if getattr(self, "cosmological_simulation", False):
             # this dataset is cosmological, so add cosmological units.
             self.unit_registry.modify("h", self.hubble_constant)
-            # Comoving lengths
-            for my_unit in ["m", "pc", "AU", "au"]:
-                new_unit = "%scm" % my_unit
-                my_u = Unit(my_unit, registry=self.unit_registry)
-                self.unit_registry.add(
-                    new_unit,
-                    my_u.base_value / (1 + self.current_redshift),
-                    dimensions.length,
-                    "\\rm{%s}/(1+z)" % my_unit,
-                    prefixable=True,
-                )
-            self.unit_registry.modify("a", 1 / (1 + self.current_redshift))
+            if getattr(self, "current_redshift", None):
+                # Comoving lengths
+                for my_unit in ["m", "pc", "AU", "au"]:
+                    new_unit = f"{my_unit}cm"
+                    my_u = Unit(my_unit, registry=self.unit_registry)
+                    self.unit_registry.add(
+                        new_unit,
+                        my_u.base_value / (1 + self.current_redshift),
+                        dimensions.length,
+                        "\\rm\{%s\}/(1+z)" % my_unit,
```

Comment:
I reverted this because the suggested change produced a flake8 error:
```
E999 SyntaxError: f-string: single '}' is not allowed
```
I tried running it through `flynt` and it had no suggestions.

Reply:
if flynt is happy, I'm happy.